tls/s2n_resume.c

```diff
@@ -155,3 +225,64 @@ int s2n_store_to_cache(struct s2n_connection *conn)
 
     return 0;
 }
+
+int s2n_connection_set_session(struct s2n_connection *conn, const uint8_t *session, size_t length)
+{
+
+    notnull_check(conn);
+    notnull_check(session);
+    int ret_val = 0;
+
+    struct s2n_blob session_data;
+    GUARD(s2n_alloc(&session_data, length));
+    memcpy(session_data.data, session, length);
```

Comment:
memcpy_check?

Reply:
memcpy_check guard returns -1 in case of failure and if that happens we will miss freeing session_data blob so used memcpy.